### ISSUE TITLE
Installs examples into lib/EXAMPLE

### ISSUE THREAD

**Issue by yurivict** [xiaoyeli/superlu_dist#111]
Usually examples are installed into ```share/examples/{project-name}```.

**Comment by xiaoyeli** [comment]
@liuyangzhuan 
I forgot why we decided to install EXAMPLE/ in lib/ ?  Shall we create another dir for examples ?

**Comment by liuyangzhuan** [comment]
They are installed in lib because spack can easily find them when doing the tests. I suggest we leave them there.

**Comment by yurivict** [comment]
lib/EXAMPLES is a very non-standard location for examples that would conflict with all linux packaging systems.
Spack should be notified that they should look in ```share/examples/{project-name}```.

**Comment by yurivict** [comment]
Spack responded that they don't assume this examples directory:
```
Spack does not care where example code is installed to, there are no customs in Spack of using that particular directory. Spack simply executes the build system of a package and lets the package decide where it should be installed. The superlu-dist package is currently hard-coded to run tests in that directory, but that can easily be changed, and even changed depending on the version of superlu-dist. So I don't think there's anything on Spack's side that is forcing superlu-dist to behave this way.
```

**Comment by liuyangzhuan** [comment]
No. The fact that spack doesn't care about the example code is exactly why we did this. We don't want to add a separate CMakeList.txt or makefile to build examples at the "spack test" stage. In the past, it caused many troubles for us... Instead, we want to generate an example executable during "spack install". From my understanding, the only place we can fetch things built from spack install is "self.prefix.lib", that's why we prefer to install examples to self.prefix.lib/EXAMPLE.
